src/relayer/Relayer.ts

```diff
@@ -304,6 +306,107 @@ export class Relayer {
     this.clients.tokenClient.decrementLocalBalance(deposit.destinationChainId, deposit.destinationToken, fillAmount);
   }
 
+  // Strategy for requesting refunds: Query all refunds requests, and match them to fills.
+  // The remaining fills are eligible for new requests.
+  async requestRefunds(sendRefundRequests = true): Promise<void> {
+    const { multiCallerClient, ubaClient } = this.clients;
+    const spokePoolClients = Object.values(this.clients.spokePoolClients);
+
+    const eligibleFillsByRefundChain: { [chainId: number]: FillWithBlock[] } = Object.fromEntries(
+      spokePoolClients.map(({ chainId }) => [chainId, []])
+    );
+
+    // For each refund/repayment Spoke Pool, group its set of by corresponding destination chain.
+    // Bound the event range by the relayer deposit lookback.
+    const refundRequestsByDstChain = Object.fromEntries(
+      spokePoolClients.map(({ chainId, latestBlockNumber }) => {
+        const lookback = latestBlockNumber - this.config.maxRelayerLookBack;
+        const refundRequests = ubaClient
+          .getRefundRequests(chainId, this.relayerAddress)
+          .filter(({ blockNumber }) => blockNumber > lookback);
+        return [chainId, refundRequests];
+      })
+    );
```

Comment:
Would this be more readable as a `reduce`?
```suggestion
    const refundRequestsByDstChain = spokePoolClients.reduce((acc, {chainId, latestBlockNumber}) => (
        {
            ...acc,
            [chainId]: ubaClient
                             .getRefundRequests(chainId, this.relayerAddress)
                             .filter(({blockNumber}) => 
                                 blockNumber > (latestBlockNumber - this.config.maxRelayerLookback)
                             )
        }
    ), {});
```